### PR TITLE
Editor: added a goto button

### PR DESCRIPTION
**Pull request by archimede67**
This feature was also suggested by Pulsar. It adds a button to go to a specified coordinate point by inputting x and y coordinates through a popup window:

image

The two number inputs are constrained between 0 and the width/height of the map (minus 1). When clicking "Go", it focuses the camera at the center of the tile at these coordinates.

## Checklist

- [x] Tested the change ingame
- [x] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [x] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/game/editor/editor.h [hunk 1]
```diff
@@ -1235,6 +1239,9 @@ class CEditor : public IEditor
 
 	unsigned char m_SwitchNum;
 	unsigned char m_SwitchDelay;
+
+	unsigned char m_GotoX;
+	unsigned char m_GotoY;
```
Comment (Chairn) [inline src/game/editor/editor.h:1244]: you mean int? Because that limits goto to [0; 255]

Comment (archimede67) [reply src/game/editor/editor.h:1244]: Yes, my bad

**Comment by archimede67** [comment]
I must admit that I did not test with a tile layer having a bigger width than the game layer, but maybe we do not need the clamp.
I didn't think of a group having an offset at all too, but then what is considered 0,0 on a group/layer having an offset? Here 0,0 is the coordinate 0,0 of the world, and this goto feature is not bound to a particular group or layer.

**Comment by heinrich5991** [comment]
I would say it's fine that this works in game layer coordinates. Not sure if clamping is necessary.

What do you think, what should be changed before merging this? @Jupeyy

**Comment by archimede67** [comment]
> probs works for most maps, but defs isnt perfect.

@Jupeyy what could make it perfect?
I will try to remove the clamping when I have time.

**Comment by sctt** [comment]
> I would say it's fine that this works in game layer coordinates. Not sure if clamping is necessary.

i disagree with that, i think that's better if coordinates are related to the current selected layer (that's coherent with coordinates that are displayed in the top-right corner, why those haven't been bound to just the game layer?).

consider that some cli tools, like map_create_pixelart, require coordinates that are relative to the selected layer, that's cool as   relative coords are displayed on the editor, so it would be useful also to jump to those coordinates with goto.

in any case, if you really want to bind goto to the gamelayer only (and i honestly don't understand whats the advantage of that), it sure mustn't clamp coordinates: other layers might have different sizes compared to the game layer, i might even want to be able to use negative coords too as i might have quads that are visualized outside of the main area (that's how map_replace_area reasons, as it works on multiple layers and groups, it asks for game layers coords as a reference, but it's just a reference, users can input anything they want to).

**Comment by archimede67** [comment]
Also, I used `INT_MAX` for the min and max as there is no real way to remove the clamping the way I did it.

File: src/game/editor/popups.cpp [hunk 2]
```diff
@@ -1800,8 +1793,8 @@ int CEditor::PopupGoto(CEditor *pEditor, CUIRect View, void *pContext)
 	};
 
 	CProperty aProps[] = {
-		{"X", pEditor->m_GotoX, PROPTYPE_INT_STEP, 0, MaxX},
-		{"Y", pEditor->m_GotoY, PROPTYPE_INT_STEP, 0, MaxY},
+		{"X", pEditor->m_GotoX, PROPTYPE_INT_STEP, -INT_MAX, INT_MAX},
+		{"Y", pEditor->m_GotoY, PROPTYPE_INT_STEP, -INT_MAX, INT_MAX},
```
Comment (Chairn) [inline src/game/editor/popups.cpp:1797]: INT_MIN ?

Comment (archimede67) [reply src/game/editor/popups.cpp:1797]: Don't you want to have the largest range possible?

**Comment by heinrich5991** [comment]
bors r+

Thanks.

**Pull request merged**